File: leet_easy_python/leet_101/leet_101a.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isSymmetric(self, root):
        if not root: return True
        return self.iSh(root.left, root.right)
        """
        :type root: TreeNode
        :rtype: bool
        """
        
    def iSh(self, left, right): # iSh = isSymmetric helper function
        if not left and not right: return True
        if not left or not right: return False
        if left.val != right.val: return False
        return self.iSh(left.left, right.right) and self.iSh(left.right, right.left)
        """
        :type left: TreeNode
        :type right: TreeNode
        :rtype: bool
        """
       

File: leet_easy_java/leet_101/leet_101a.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root == null) return true;
        return iSh(root.left, root.right);
    }
    private boolean iSh(TreeNode left, TreeNode right){ //~ iSh = is Symmetric helper
        if(left == null && right == null) return true;
        if(left == null || right == null || left.val != right.val) return false;
        return iSh(left.left, right.right) && iSh(left.right, right.left);
    }
}